src/TagsManagerWithStream.js

```diff
@@ -0,0 +1,55 @@
+const TagsManager = require("./TagsManager");
+
+const DEFAULT_WRITE_CHUNK = 2000;
+
+class TagsManagerWithStream extends TagsManager {
+  constructor(stream, writeChunk = DEFAULT_WRITE_CHUNK) {
+    super();
+    this.writeChunk = writeChunk;
+    this.stream = stream;
+    this.isWriting = false;
+    this.stream.on("drain", () => this._resumeWriting());
+  }
+
+  push(code, value) {
+    super.push(code, value);
+
+    if (this.stream && this.stream.writable && !this.isWriting && this.lines.length > this.writeChunk) {
```

Comment:
If we were to make this async, could we do something like

```js
if (this.lines.length > this.writeChunkSize) {
    if (isDraining) {
        await new Promise((resolve) => {
            this.stream.on('drain', resolve);
        });
    }
    this._writeChunkToStream(...)
}
```

This would basically just only make us await while we're waiting for that drain to occur, before we continue pushing more. Problem just being that now everything that calls `push` needs to await the promise. We basically need to yield the generator 😆

Reply:
I think, as previously discussed, leave this for now?

Reply:
ok as per new discussion, let me make it async 🙇